src/providers/multicall-uniswap-provider.ts

```diff
@@ -42,13 +34,10 @@ export class UniswapMulticallProvider extends IMulticallProvider<UniswapMultical
   constructor(
     protected chainId: ChainId,
     protected provider: BaseProvider,
-    protected gasLimitPerCall = 1_000_000,
-    protected multicallAddressOverride = UNISWAP_MULTICALL_ADDRESS
+    protected gasLimitPerCall = 1_000_000
   ) {
     super();
-    const multicallAddress = multicallAddressOverride
-      ? multicallAddressOverride
-      : contractAddressByChain[this.chainId];
+    const multicallAddress = UNISWAP_MULTICALL_ADDRESSES[this.chainId];
```

Comment:
Why have we removed the functionality for overriding the address?

Reply:
Since the smart-order-router repository maintains the list of supported chain IDs I don't believe it's necessary to have an override here (it's not used anywhere). 

So I figured removing the unused code would make the constructor a bit cleaner. Let me know what you think! Happy to add it back

Reply:
I've also considered the prospect of the next chains integrating Uniswap, contributing to the smart-order-router is a pre-req and they'll have to add their unique [chainId: Multicall address] regardless